### PR TITLE
SerialPort: Cleanup ArgumentException paramNames/messages

### PR DESCRIPTION
**Pull request by justinvp**
Use `nameof`, specify the `paramName` where appropriate, and remove some unnecessary resources.

### CODE REVIEW

File: src/System.IO.Ports/src/Resources/Strings.resx [hunk 1]
```diff
@@ -100,12 +100,6 @@
   <data name="Argument_InvalidOffLen" xml:space="preserve">
     <value>Offset and length were out of bounds for the array or count is greater than the number of elements from index to the end of the source collection.</value>
   </data>
-  <data name="ArgumentNull_Array" xml:space="preserve">
-    <value>Array cannot be null.</value>
-  </data>
-  <data name="ArgumentNull_Buffer" xml:space="preserve">
-    <value>Buffer cannot be null.</value>
-  </data>
```
Comment (justinvp) [inline src/System.IO.Ports/src/Resources/Strings.resx]: These two resources don't really add much value over the default `ArgumentNullException` message.

File: src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs [hunk 2]
```diff
@@ -1145,14 +1145,14 @@ internal unsafe void Write(byte[] array, int offset, int count, int timeout)
...
-                throw new ArgumentNullException(nameof(array), SR.ArgumentNull_Array);
+                throw new ArgumentNullException(nameof(array));
             if (offset < 0)
                 throw new ArgumentOutOfRangeException(nameof(offset), SR.ArgumentOutOfRange_NeedPosNum);
             if (count < 0)
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedPosNum);
             if (count == 0) return; // no need to expend overhead in creating asyncResult, etc.
             if (array.Length - offset < count)
-                throw new ArgumentException(nameof(count), SR.ArgumentOutOfRange_OffsetOut);
+                throw new ArgumentException(SR.Argument_InvalidOffLen);
```
Comment (justinvp) [inline src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs]: Every other place where `if (array.Length - offset < count)` is being checked is using `SR.Argument_InvalidOffLen` for the message, so use the same message here for consistency.

This was the only place where `SR.ArgumentOutOfRange_OffsetOut` was being used and the message doesn't make sense here (`"Either offset did not refer to a position in the string, or there is an insufficient length of destination character array."`).

**Comment by justinvp** [comment]
cc: @JeremyKuhne, @willdean

**Comment by willdean** [comment]
lgtm

File: src/System.IO.Ports/src/Resources/Strings.resx [hunk 3]
```diff
@@ -124,9 +118,6 @@
   <data name="ArgumentOutOfRange_WriteTimeout" xml:space="preserve">
     <value>The timeout must be either a positive number or -1.</value>
   </data>
-  <data name="ArgumentOutOfRange_OffsetOut" xml:space="preserve">
-    <value>Either offset did not refer to a position in the string, or there is an insufficient length of destination character array."</value>
-  </data>
```
Comment (danmosemsft) [inline src/System.IO.Ports/src/Resources/Strings.resx]: Incidentally I notice some of the strings end in quotes by accident

````
<value>Can not access a closed Stream."</value>		     
<value>Argument {0} cannot be null or zero-length."</value>		     
```

**Comment by danmosemsft** [comment]
LGTM

**Pull request merged**